API_doc/Overview.md

```diff
@@ -0,0 +1,59 @@
+# Creating and retrieving retrieving a configuration
+
+The first call $LIB_NAME should be to the following function:
+```cpp
+$LIB_NAME::Config $LIB_NAME::createConfig(char const *configName);
+```
+`configName` the name given to the configuration.
+
+```cpp
+struct $LIB_NAME::Id
+{
+  void *data;
+  size_t size; // size of data
+};
+
+$LIB_NAME::Id $LIB_NAME::Config::getId();
+$LIB_NAME::Config $LIB_NAME::getConfig($LIB_NAME::Id id);
+```
+
+Using the above functions it's possible for an application to persistently use one configuration once it has been created.
+
+# Modifying a configuration
+
+A configuration is represented by a series of instructions.
+The most basic instruction is a setting, which can alias another setting.
+
+```cpp
+void $LIB_NAME::Config:addSetting(char const *name, char const *value);
+void $LIB_NAME::Config:addSettingAlias(char const *name, char const *aliasName);
+```
+
+It is also possible to apply all settings from another `$LIB_NAME::Config`
+
+```cpp
+void $LIB_NAME::Config:importConfig($LIB_NAME::Config config);
```

Comment:
Par contre, je comprends pas l'intérêt de cette fonction.

Reply:
For config inheritence: basic idea being that at least most configuration will inherit from a global configuration. Maybe `import` isn't the best name and `include` would be clearer?

Reply:
Ok, I think I'm not sure of what you want to do
Two possibilities :
- this func has for goal to "clone" a configuration, so I think something like `cloneConfig` can be better
- this func has for goal to make a kind of inheritance (and if the father is modified, the child too), so "includeConfig" or "inheritFrom" could be clearer

Reply:
I meant the second. I'm okay with both include and import but not clone. Go ahead and rename to what you find clearer.